src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -238,7 +238,12 @@
     <Field Name="MethodOpt" Type="MethodSymbol" Null="allow"/>
     <Field Name="ResultKind" PropertyOverrides="true" Type="LookupResultKind"/>
   </Node>
-  
+
+  <Node Name="BoundNotNullExpression" Base="BoundExpression">
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="allow"/>
```

Comment:
I believe the principal intended semantics are to suppress a warning, not make the compiler think the value is not null. At least, that's the latest view. Perhaps BoundSuppressNullableWarningExpression?

Reply:
FWIW if this could be used on `Nullable<T>` it's not just suppressing warnings. it force-unwrap the value.

Reply:
We can rename the node again if the scope or meaning change.